unit-tests/unit-tests-hdr.cpp

```diff
@@ -309,276 +309,312 @@ TEST_CASE("HDR Config while Streaming", "[hdr][live][using_pipeline]")
 
-        pipe.stop();
+                    seq_id = merged_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_SEQUENCE_ID);
+                    frame_counter = merged_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_FRAME_COUNTER);
+                    frame_exposure = merged_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_EXPOSURE);
+                    frame_ts = merged_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_FRAME_TIMESTAMP);
+                    //std::cout << "merged - seq_id = " << seq_id << ", frame counter = " << frame_counter << ", exposure = " << frame_exposure << ", ts = " << frame_ts << std::endl;
 
-        //std::cout << "------------------stop - start again ---------------" << std::endl;
+                    if (i == (num_of_iterations_in_serie - 1))
+                        first_series_last_merged_ts = frame_ts;
+                }
+            }
 
-        depth_sensor.set_option(RS2_OPTION_HDR_ENABLED, 1);
-        REQUIRE(depth_sensor.get_option(RS2_OPTION_HDR_ENABLED) == 1.f);
+            pipe.stop();
 
-        pipe.start(cfg);
+            //std::cout << "------------------stop - start again ---------------" << std::endl;
 
-        for (int i = 0; i < 10; ++i)
-        {
-            rs2::frameset data = pipe.wait_for_frames();
-            rs2::depth_frame out_depth_frame = data.get_depth_frame();
-
-            auto seq_id = out_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_SEQUENCE_ID);
-            long long frame_counter = out_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_FRAME_COUNTER);
-            long long frame_exposure = out_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_EXPOSURE);
-            long long frame_ts = out_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_FRAME_TIMESTAMP);
-            //std::cout << "depth - seq_id = " << seq_id << ", frame_counter = " << frame_counter << ", exposure = " << frame_exposure << ", ts = " << frame_ts << std::endl;
-
-            // merging the frames from the different HDR sequence IDs 
-            auto merged_frameset = merging_filter.process(data);
-            auto merged_depth_frame = merged_frameset.as<rs2::frameset>().get_depth_frame();
-
-            seq_id = merged_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_SEQUENCE_ID);
-            frame_counter = merged_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_FRAME_COUNTER);
-            frame_exposure = merged_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_EXPOSURE);
-            frame_ts = merged_depth_frame.get_frame_metadata(RS2_FRAME_METADATA_FRAME_TIMESTAMP);
-            //std::cout << "merged - seq_id = " << seq_id << ", frame_counter = " << frame_counter << ", exposure = " << frame_exposure << ", ts = " << frame_ts << std::endl;
-            REQUIRE(frame_ts > first_series_last_merged_ts);
-        }
+            depth_sensor.set_option(RS2_OPTION_HDR_ENABLED, 1);
```

Comment:
Not clear what should be the state of HDR after pipe.stop() while the depth_sensor handle is still active:
- if it is reset then check that its value was 0 before setting 1. Otherwise  check that it is already on

Reply:
Done